src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/AutonomousManager.java

```diff
@@ -120,4 +128,79 @@ public SortedMap<String, String> getAllScanSettingsProperties() {
 
         return scanSettingsProperties;
     }
+
+    public void updateScanSettingsProperties(SortedMap<String, String> propertiesMap, List<String> adoptedScanTypes) {
+        userProvidedProperties.forEach((propertyKey, propertyValue) -> {
+            Optional<String> scanTypeValue = findScanType(adoptedScanTypes, propertyKey);
+            determinePropertyTypeAndUpdate(propertyKey, propertyValue, scanTypeValue, true);
+        });
+        propertiesMap.forEach((propertyKey, propertyValue) -> {
+            Optional<String> scanTypeValue = findScanType(adoptedScanTypes, propertyKey);
+            determinePropertyTypeAndUpdate(propertyKey, propertyValue, scanTypeValue, false);
+        });
+
+        scanSettings.setDetectorSharedProperties(detectorSharedProperties);
+        scanSettings.setGlobalDetectProperties(globalProperties);
+    }
+
+    private void determinePropertyTypeAndUpdate(String propertyKey, String propertyValue, Optional<String> scanTypeValue, boolean userProvidedProperty) {
+        if(isDetectorTypeProperty(propertyKey, propertyValue)) {
+            updateDetectorProperties(propertyKey, propertyValue, userProvidedProperty);
+        } else if (scanTypeValue.isPresent()) {
+            String scanTypeName = scanTypeValue.get();
+            ScanType scanType = scanSettings.getScanTypeWithName(scanTypeName);
+            if(userProvidedProperty && !propertyValue.isEmpty()) {
+                scanType.getScanProperties().put(propertyKey, propertyValue);
+            } else if(!scanType.getScanProperties().containsKey(propertyKey) && !propertyValue.isEmpty()) {
+                scanType.getScanProperties().put(propertyKey, propertyValue);
+            }
+        } else if(isGlobalTypeProperty(propertyKey, propertyValue)) {
+            updateGlobalProperties(propertyKey, propertyValue, userProvidedProperty);
+        }
+    }
+
+    private Optional<String> findScanType(List<String> adoptedScanTypes, String key) {
+        return adoptedScanTypes.stream().filter(tool -> {
+            String tempKey = key.replace(".","_");
+            return tempKey.contains(tool.toLowerCase());
+        }).findFirst();
+    }
+
```

Comment:
This functions helps in finding if the key contains any tool or package manager in it from all the adopted scan types. Example: If BINARY_SCAN was ran and detect.binary.scan.file.path would fall in such category. Similarly, if GRADLE package manager was ran, then detect.gradle.path applies here.